tests/lib/hardware/hardware_base.py

```diff
@@ -79,6 +80,17 @@ def generate_keys(self):
             "%s%s_key" % (config.CLUSTER_PREFIX, self.hardware_uuid)
         self.pubkey = "%s %s" % (key.get_name(), key.get_base64())
 
+    def ssh_agent(self):
+        res = subprocess.run(['ssh-agent'], capture_output=True)
+        self.SSH_AUTH_SOCK = res.stdout.decode(
+            'utf-8').split(';')[0].split('=')[1]
+        self.SSH_AGENT_PID = res.stdout.decode(
+            'utf-8').split(';')[2].split('=')[1]
+        os.environ['SSH_AUTH_SOCK'] = self.SSH_AUTH_SOCK
+        os.environ['SSH_AGENT_PID'] = self.SSH_AGENT_PID
+        res = subprocess.run(['ssh-add', self.private_key],
```

Comment:
please use `check_call` or something that does check the return code.

Reply:
Add check for result.